core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java

```diff
@@ -143,12 +144,9 @@ public ListBucketResult(
         //marker filter
         .filter((status) -> (status.getPath().compareTo(marker) > 0)
             //prefix filter
-            && status.getPath().startsWith(prefix + mPrefix)
-            //folder filter
-            && ((mDelimiter != null && mDelimiter.equals(AlluxioURI.SEPARATOR))
```

Comment:
@jffree I don't think we need this check anymore since we change the way for us to list status, can you help confirm this?

Reply:
I think this makes sense; you're supposed to be able to have a `null` delimiter. I also don't know why there was a "folder filter" for `ListBucketResult`.

Reply:
```
aws s3api list-objects-v2 --endpoint-url http://localhost:39999/api/v1/s3 --bucket shouwei-test
{
    "Contents": [
        {
            "LastModified": "2021-11-18T20:04:04.068Z", 
            "Key": "alluxio", 
            "Size": 1
        }, 
        {
            "LastModified": "2021-11-18T20:04:04.068Z", 
            "Key": "alluxio/special_backups", 
            "Size": 1
        }, 
        {
            "LastModified": "2021-11-03T18:33:08.000Z", 
            "Key": "alluxio/special_backups/alluxio-backup-2021-11-04-1635989588831.gz", 
            "Size": 190
        }
    ]
}
aws s3api list-objects-v2 --bucket shouwei-test
{
    "Contents": [
        {
            "LastModified": "2021-11-04T01:33:07.000Z", 
            "ETag": "\"d41d8cd98f00b204e9800998ecf8427e\"", 
            "StorageClass": "STANDARD", 
            "Key": "alluxio/", 
            "Size": 0
        }, 
        {
            "LastModified": "2021-11-04T01:33:08.000Z", 
            "ETag": "\"d41d8cd98f00b204e9800998ecf8427e\"", 
            "StorageClass": "STANDARD", 
            "Key": "alluxio/special_backups/", 
            "Size": 0
        }, 
        {
            "LastModified": "2021-11-04T01:33:08.000Z", 
            "ETag": "\"4a6b70de44d18dffbd94860a7d60f9e2\"", 
            "StorageClass": "STANDARD", 
            "Key": "alluxio/special_backups/alluxio-backup-2021-11-04-1635989588831.gz", 
            "Size": 190
        }
    ]
}
```
Verified, thanks @jffree